scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1342,8 +1363,13 @@
                                 :init-container-checkpoint-volume-mounts init-container-checkpoint-volume-mounts
                                 :init-container-workdir init-container-workdir
                                 :init-container-workdir-volume-mount-fn init-container-workdir-volume-mount-fn
+                                :job job
```

Comment:
This abstraction (stuffing a lot of local variables into the container-attributes dictionary) seems undesirarable. 

* There's an unnecessary risk of typo's; mistype a key here or in the consumer and the output is nil. Accessing the variables under their original name would catch those kinds of typo errors.
* There's redundancy here; the job and job-name are put in here twice, pool-name is put here twice. the user is put here twice.)
* Its harder to follow which consumer is using which variables.

If we wan to split it up into subfunctions, why not (let [subfn (fn add-container-to-pod ....)] and let it grab the context (user, pool, etc) from the environment?

Reply:
I agree but want to make a separate PR

Reply:
I don't think it would take much to do that. Just copy&paste the function body and look for undefined vars. I'm not a super fan of perpetuating this further......